src/update_agent/actor.rs

```diff
@@ -381,27 +394,39 @@ impl UpdateAgent {
         release: Release,
     ) -> ResponseActFuture<Self, Result<Release, ()>> {
         if !can_finalize {
+            // Reset number of postponements if finalization attempt failed due to
+            // strategy constraints.
+            update_unit_status(&format!(
+                "update staged: {}; reboot pending due to update strategy",
+                &release.version
+            ));
+            self.state.update_staged(release);
             return Box::pin(actix::fut::err(()));
         }
 
-        // Warn logged in users of imminent reboot.
-        let msg = format!(
-            "staged deployment '{}' available, proceeding to finalize it and reboot",
+        log::info!(
+            "staged deployment '{}' available, proceeding to finalize it",
             release.version
         );
-        log::info!("{}", &msg);
-        match broadcast(&msg) {
-            Ok((sessions_total, sessions_broadcasted)) => {
-                if sessions_total != sessions_broadcasted {
-                    log::warn!(
-                        "{} sessions found, but only broadcasted to {}",
-                        sessions_total,
-                        sessions_broadcasted
-                    );
+
+        // Postpone reboot if there are interactive users logged in.
+        match super::get_interactive_user_sessions() {
+            Ok(interactive_sessions) => {
+                DETECTED_ACTIVE_USERS.set(interactive_sessions.len().try_into().unwrap());
```

Comment:
I recall in a previous PR, @lucab suggested casting the `usize` here `as i64`. However, wouldn't this cast potentially lead to data loss since `usize` is likely a `u64`? This time, I've opted to use `std::convert::TryInto::try_into()`, but let me know if `as` would actually work just fine.

Reply:
Yes, it's a minor concern (you'll never see 2⁶³ user sessions), but technically right.
An `as` could be good enough.
Or you can write a small `saturating_as` helper which clamps the input to the proper output range, before doing the conversion.